### PR TITLE
Beginning to improve code quality accross hydration functions

### PR DESCRIPTION
**Pull request by tommusrhodus**
#### Changes proposed in this Pull Request

* Beginning work on improving code quality.
* Improving docblocks
* Removing direct global calls
* Improves parse context

#### Testing instructions

* This will not pass automated PHPCS checks currently.

### CODE REVIEW

**Comment by tommusrhodus** [comment]
Oh, it did pass checks 🤷

File: .phpunit.result.cache
Comment (aaronjbaptiste) [inline .phpunit.result.cache:1]: I know next to 0 about php testing, but should this cache file be gitignored ?

**Review by aaronjbaptiste — APPROVED** [review]
Appreciate the documentation updates!

File: .phpunit.result.cache
Comment (tommusrhodus) [reply .phpunit.result.cache:1]: Will add it in my next PR for sure.

**Pull request merged**